### ISSUE TITLE
DeprecationWarning: PKCS#12 support in pyOpenSSL is deprecated

### ISSUE THREAD

**Issue by kkirsche** [m-click/requests_pkcs12#27]
Thank you for the great package, I'm reporting the following deprecation warning that is triggered within one of my applications:

```
/Users/REDACTED/Library/Caches/pypoetry/virtualenvs/REDACTED-PACKAGE-2-MN45-t-py3.9/lib/python3.9/site-packages/requests_pkcs12.py:20: DeprecationWarning: PKCS#12 support in pyOpenSSL is deprecated. You should use the APIs in cryptography.
  from OpenSSL.crypto import load_pkcs12, dump_certificate, dump_privatekey, FILETYPE_PEM
```

I have not had a chance to look into how difficult this would be to address but wanted to let you know in case you were not aware.

Thanks again for the great package, I hope to take a look at this and see if I may be able to contribute a fix, I know it can be frustrating to get bug reports with no solutions or help. If you are interested, please let me know so as to not create any confusion

**Comment by vog** [comment]
Thanks for the hint!

Of course, patches that fix warnings, deprecation warnings as well as any other warnings, are always welcome!

**Comment by kkirsche** [comment]
Reference note for myself (and others curious) which will be edited as I find the newer documentation. 

Old-ish (3.1 not 3.4.7) documentation reference for cryptography:
https://cryptography.io/en/3.1/_modules/cryptography/hazmat/primitives/serialization/pkcs12/

Latest:
https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization/#pkcs12

**Comment by kkirsche** [comment]
The only thing I seem to be getting hung up on is the private key dumping in `create_ssl_sslcontext` — but will take a look into it later.  I'll make a work-in-progress (WIP) pull request (PR) to share what I've been doing.

**Comment by kkirsche** [comment]
Created https://github.com/m-click/requests_pkcs12/pull/28 for you to take a look at what I've started with and to see what I'm missing. It's a broken fragment currently still, so bear with me, please, as it's a bit rough still.

**Comment by vog** [comment]
I had a look at this and it looks okay, so far. Do you need any further feedback?

**Comment by ao-landoo** [comment]
We have an application which is also triggering deprecation warning. Hopefully you can release a new version of the library soon.

**Comment by kkirsche** [comment]
> I had a look at this and it looks okay, so far. Do you need any further feedback?

The server I was going to be using to test the intermediate work sadly went away so I don’t have  a way to test the change set moving forward. I’d love to finish it I just don’t have a way to test that it’s functional. If you have a suggestion though I’d love to hear how you approach this

**Comment by vog** [comment]
@ao-landoo @kkirsche I'd propose to connect with each other for further testing.

**Comment by ao-landoo** [comment]
I run a test on the application I'm using and it LOOKS LIKE it's working fine.

**Comment by vog** [comment]
@ao-landoo Thanks!

@kkirsche Would you like to make any further improvements to your pull request, or do you consider it being ready for review?

**Comment by ao-landoo** [comment]
@kkirsche Could you answer to @vog's question please?

**Comment by kkirsche** [comment]
There’s more that could and probably should be done but I won’t have the availability in the timeframe desired to do it. If you’re comfortable with the functionality of it @vog than I’d say roll with it. I’d recommend squashing the commits though on merge in case something was missed and you need to revert it.

**Comment by vog** [comment]
Okay, I think we can take the code mostly as-is, squashed into a single commit, but I need to fix some minor coding style issues, and more importantly, need to declare the PIP dependency on the [cryptography package](https://pypi.org/project/cryptography/).

Moreover, I assume this is also the time to drop support for Python 2, but that should be a [no-brainer](https://www.python.org/doc/sunset-python-2/), at least for any slightly security-sensitive application.

**Comment by vog** [comment]
This evening I'll take care of this, and will also create a new release 1.11.

**Comment by vog** [comment]
I just pushed an improved version of @kkirsche's pull request #28:

* Commit cb09e98bc4a6aa4d7c776822a6f59c9ed422f735

I also officially removed Python 2 support (factually this already happened by the above commit), together with some more packaging improvements. Finally, I released version 1.11:

* https://pypi.org/project/requests-pkcs12/
